src/about-this-guide.md

```diff
@@ -6,16 +6,16 @@ development.
 
 There are six parts to this guide:
 
-1. [Contributing][p1]: Contains information that should be useful no matter how
+1. [Building, Debugging, and Contributing to `rustc`][p1]: Contains information that should be useful no matter how
    you are contributing, such as procedures for contribution, building the
    compiler, etc.
-2. [High-level Compiler Architecture][p2]: Discusses the high-level
-   architecture of the compiler, especially the query system.
-3. [The Compiler Frontend][p3]: Discusses the compiler frontend and internal
-   representations.
-4. [The Type System][p4]: Discusses the type system.
-5. [The Compiler Backend][p5]: Discusses the compiler backend, code generation,
-   linking, and debug info.
+2. [How `rustc` Works][p2]: Discusses the high-level
+   architecture of the compiler and stages of the compile process
+3. [Source Code Representation][p3]: Describes the process of taking raw source code from the user and
+   transforming it into various forms that the compiler can work with easily
+4. [Analysis][p4]: discusses the analyses that the compiler uses to check various
+   properties of the code and inform later stages of the compile process
+5. [From MIR to Binaries][p5]: How linked executable machine code is generated
```

Comment:
Minor nit: can you end each point with a period?

Reply:
No problem!  Will add them now.

Reply:
978d894938f0d0990df1b82c4b72ee77381625f6